pennylane/workflow/qnode.py

```diff
@@ -249,12 +259,24 @@ def _validate_gradient_kwargs(gradient_kwargs: dict) -> None:
 
 
 def _validate_qfunc_output(qfunc_output, measurements) -> None:
-    if isinstance(qfunc_output, qml.numpy.ndarray):
-        measurement_processes = tuple(measurements)
-    elif not isinstance(qfunc_output, Sequence):
-        measurement_processes = (qfunc_output,)
+    measurement_processes = qml.pytrees.flatten(
+        qfunc_output,
+        is_leaf=lambda obj: isinstance(obj, qml.measurements.MeasurementProcess),
+    )[0]
+
+    # user provides no measurements or non-measurements
+    if len(measurement_processes) == 0:
+        measurement_processes = None
     else:
-        measurement_processes = qfunc_output
+        # patch for tensor measurement objects, e.g., qml.math.hstack <-> [tensor([tensor(...), tensor(...)])]
+        if isinstance(measurement_processes[0], Iterable) and any(
+            isinstance(m, qml.typing.TensorLike) for m in measurement_processes[0]
+        ):
+            measurement_processes = [
+                m.base.item()
+                for m in measurement_processes[0]
+                if isinstance(m.base.item(), qml.measurements.MeasurementProcess)
+            ]

```

Comment:
@albi3ro: This feels quite hacky, is there a way to register a `qml.numpy.tensor.tensor` into the PyTree so that it knows how to flatten through a tensor of measurement operators? I tried doing something similar to the `autograd` registration but I can't get around circular import erros. 😄

Reply:
I'm ok with this patch.  We don't really want a `qml.numpy.tensor` to be in pytree in any case other than this one.